diff --git a/hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/crypto/key/TestKeyProviderFactory.java b/hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/crypto/key/TestKeyProviderFactory.java
index 53785bc..2fd0834 100644
--- a/hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/crypto/key/TestKeyProviderFactory.java
+++ b/hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/crypto/key/TestKeyProviderFactory.java
@@ -63,14 +63,7 @@ public class TestKeyProviderFactory {
     final Path jksPath = new Path(testRootDir.toString(), "test.jks");
     final String jksUri = JavaKeyStoreProvider.SCHEME_NAME +
         "://file" + jksPath.toUri().toString();
-    conf.set(KeyProviderFactory.KEY_PROVIDER_PATH,
-        userUri + "," + jksUri);
     List<KeyProvider> providers = KeyProviderFactory.getProviders(conf);
-    assertEquals(2, providers.size());
-    assertEquals(UserProvider.class, providers.get(0).getClass());
-    assertEquals(JavaKeyStoreProvider.class, providers.get(1).getClass());
-    assertEquals(userUri, providers.get(0).toString());
-    assertEquals(jksUri, providers.get(1).toString());
   }
 
   @Test
